### ISSUE TITLE
RMSE of p(z)

### ISSUE THREAD

**Issue by cavuoti** [LSSTDESC/PZDC1paper#14]
I'm not sure that the RMSE tells us much about the individual PDFs. 
In the first place we should define what we want to consider as the pdf of the true redshift, a pdf in which all the values are 0 with the exception of the bin of zspec? or a gaussian pdf centered on zpsec and a width related to the error?
But in any case I'm not convinced that the average of those values tells something about the pdfs.

On the other hand we could perform it on the stacked PDFs, but again we should decide what is the pdf of true redshift would be, and in principle I think that other indicator could be better for such purpose, what is your opinion?

**Comment by aimalz** [comment]
@cavuoti I agree that the RMSE doesn't mean anything without a notion of "truth," and a delta function at z_spec isn't a true photo-z PDF because it's conditioned on different data.  We could use it to compare the stacked n(z) to the true n(z), but that's not really a fair comparison either because assigning every galaxy a PDF equal to the distribution of training set redshifts would "win" by this metric.  (Actually, I think that strategy might win by all our metrics, but I'm not sure if the DC1 paper is the right place to bring it up.)  Most of our metrics seem to be on the PIT because it should be flat, so this could potentially be one of those if people do indeed want to include it.

**Comment by sschmidt23** [comment]
I agree that the RMSE for the individual p(z)'s does not tell us much here, as we do not have a "true" p(z) to compare against.  For the stacked p(z) vs true N(z), I think it could still have value as a comparative metric between the codes.  The fact that such a problem exists, where the individual p(z)'s can be incorrect but the ensembles that we are testing still indicate no problems, should be pointed out in the paper draft, though.  

For the simple test of all galaxies with i<25.3 and representative training data that we are using for DC1, as Alex says, a code that just returned p(z) = normalized N(z) of the training sample would give great results.  But, this will not be true if we tested tomographic bins, different magnitude slices, or if we had incomplete training data.  I believe that during a recent telecon we discussed inclusion of tomographic bins, but decided that it deserved a separate follow up paper.  

My opinion is that we look at RMSE for the stacked distribution of p(z) compared to spec-z N(z) (note the same bandwidth smoothing question raised for KLD in issue [#13](https://github.com/LSSTDESC/PZDC1paper/issues/13) enters here as well).  I would say that we should mention the potential failure modes of the test, and also use them as motivation for the follow up paper and DC2 work on incompleteness in Future Work.

**Comment by janewman-pitt-edu** [comment]
The interpretation of a given value of RMSE will depend on the width of the PDF (or n(z) ), so I definitely don't think it's a great statistic here.  (i.e.: being off by 0.01 on average in a PDF that is nonzero from 0.99 to 1 is very different than in one that is nonzero from 0 to 4).

**Comment by aimalz** [comment]
This was resolved a while back, so I'm closing the issue.
